origami/origami.py

```diff
@@ -226,12 +240,47 @@ def _wrapper():
             return response
         return _wrapper
 
-    def send_text_array(self, data):
+    def _origmai_send_data(self, data, dataType):
+        """
+        Core function which sends output to either the origami server or the
+        user as response to request
+
+        Args:
+            data: list or tuple of string to be sent.
+            dataType: Key for data in payload python dict
+                can be either of `data` or `terminalData`
+
+        Returns:
+            resp: Response we sent to user as API response or response from
+                the origami server.
+        """
+        resp = None
+        socketId = user_req.form.get(constants.REQUEST_SOCKET_ID_KEY, type=str)
+        # Check if a valid socketId is provided in the request
+        # else consider it as an API request.
+        if socketId:
+            # Check if the socket-id is there is the request form.
+            payload = {
+                "socketId": socketId,
+                dataType: data
+            }
+            resp = self.request_origami_server(payload)
+        else:
+            # TODO: Discuss the strucutre of API response payload.
```

Comment:
When can we do that?

Reply:
Currently, I am sticking to the earlier format for the data format used in origami frontend so there won't be many changes on the frontend. Once we are done with the basic implementation of everything we can change them very easily after a discussion.